src/components/BlogForm/BlogForm.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { Form, Field } from "react-final-form";
+import { useDispatch } from "react-redux";
+import { addBlogPostAndLocalStorage } from "./../../actions/blogActions";
+import ImageDnD from "../ImageDnD/ImageDnD";
+import SelectBox from "../SelectBox/SelectBox";
+import "./BlogForm.scss";
+
+const BlogForm = () => {
+  const dispatch = useDispatch();
+  const [imagePreview, setImagePreview] = useState(null);
+  const [selectedTags, setSelectedTags] = useState([]);
+
+  const onSubmit = async (values, form) => {
+    const blog = { ...values, image: imagePreview, tags: selectedTags };
+    await dispatch(addBlogPostAndLocalStorage(blog));
+    setImagePreview(null);
+    form.reset();
+  };
+
+  const handleDrop = (acceptedFiles) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const base64Image = reader.result;
+      setImagePreview(base64Image);
+    };
+    reader.readAsDataURL(acceptedFiles[0]);
+  };
+
+  const cancelImagePreview = () => {
+    setImagePreview(null);
+  };
+
+  const handleCancel = (form) => {
+    form.reset();
+    setImagePreview(null);
+  };
+
+  const tags = [
+    { value: "Technology", label: "Technology" },
+    { value: "Poetry", label: "Poetry" },
+    { value: "Films", label: "Films" },
+    { value: "World Politics", label: "World Politics" },
+  ];
```

Comment:
will these always be static? also we can move them to a separate file as json and use it from there

Reply:
done